include/zephyr/mgmt/mcumgr/smp/mcumgr_client.h

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef H_MGMT_MCUMGR_CLIENT_
+#define H_MGMT_MCUMGR_CLIENT_
+
+#include <inttypes.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+/**
+ * @brief MCUMGR Image Group response types.
+ */
+enum mcumgr_image_rsp_t {
+	/** Image state write or read response */
+	MGMT_IMAGE_STATE_RSP = 0,
+
+	/** Image Upload response*/
+	MGMT_IMAGE_UPLOAD_RSP,
+
+	/** Secondary image erase response */
+	MGMT_IMAGE_ERASE_RSP
+};
+
+/**
+ * @brief MCUMGR OS Group response types.
+ */
+enum mcumgr_os_rsp_t {
+	/** OS response */
+	MGMT_OS_ECHO_RSP = 0,
+
+	/** Image Upload response*/
+	MGMT_OS_RESET_RSP
+};
+
+/**
+ * @brief Image list data.
+ */
+struct mcumgr_image_data {
+	int slot_num;
+	int img_num;
+	char hash[33];
```

Comment:
there is a define somewhere in mcumgr for the sha256 hash size, and it should be 32 bytes, not 33 (null character isn't needed)

Reply:
Ok. Let me think. Hash could be also shorter than 32 in some point. That's why 1 extra byte is needed for max lenght

Reply:
Hash is either 0 if not present or 32 for an sha256 - other lengths are invalid

Reply:
I agree.